reactive/kubernetes_master.py

```diff
@@ -893,6 +898,10 @@ def set_final_status():
         hookenv.status_set('waiting', 'Waiting to apply keystone policy file.')
         return
 
+    if not is_flag_set('certificates.available'):
+        missing_certificates_notice()
```

Comment:
With this approach, `get_flags()` function is called twice (in decorator `when_not` and in function `is_flag_set`). Do you want to use the `missing_certificates_notice` function elsewhere?

Reply:
Yeah, I'm not particularly happy about this either, but as you can see, that's the general purpose of this `@atexit` function. You need to stop its execution otherwise it sets status to `active` at the end. Without this `if`, the flow would go like this:
* `certificates.available flag changes
* `missing_certificates_notice()` is executed
* `blocked` state is set
* `@atexit` function is executed
* `active` status is set

Reply:
Yes, I know that this `if` is needed, but the `missing_certificates_notice` function isn't?

- `certificates.available` flag changes
- `@atexit` function is executed
- `blocked` status is set (with `if` and `hookenv.status_set(...)`)

The correct question is whether the `atexit` function will be triggered when the `certificates.available` (or another) flag changes. Do you know if yes?

Reply:
Oh, you meant it like that...I don't think `@atexit` would be executed without explicit hook reacting to the flag change. According to the documentation `atexit: Schedule a callback to run on successful hook completion.` I can give it a shot though.

Reply:
As suspected, without the explicit hook the status wont change to `blocked`, at least not immediately. Eventually some other hook runs and the status is changed but that could take up to 5 minutes (`update-status`) and I don't think it's wise to rely on other hooks.

Reply:
Just for the sake of complete discussion:

We eventually found out that moving `certificates.available` check closer to the beginning of the `set_final_status()` would set expected status immediately after CA departure. This allowed us to remove the explicit hook.